src/main/java/ch/njol/skript/expressions/ExprSignText.java

```diff
@@ -18,159 +18,298 @@
  */
 package ch.njol.skript.expressions;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
 import org.bukkit.block.Block;
 import org.bukkit.block.Sign;
+import org.bukkit.block.sign.Side;
 import org.bukkit.event.Event;
 import org.bukkit.event.block.SignChangeEvent;
-import org.eclipse.jdt.annotation.Nullable;
+import org.jetbrains.annotations.Nullable;
+
+import com.google.common.collect.Lists;
 
 import ch.njol.skript.Skript;
-import ch.njol.skript.aliases.Aliases;
-import ch.njol.skript.aliases.ItemType;
 import ch.njol.skript.classes.Changer.ChangeMode;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.effects.Delay;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.lang.util.SimpleExpression;
 import ch.njol.skript.lang.util.SimpleLiteral;
+import ch.njol.skript.util.chat.BungeeConverter;
+import ch.njol.skript.util.chat.ChatMessages;
 import ch.njol.util.Kleenean;
+import ch.njol.util.StringUtils;
+import ch.njol.util.coll.CollectionUtils;
+import net.kyori.adventure.text.serializer.bungeecord.BungeeComponentSerializer;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Sign Text")
 @Description("A line of text on a sign. Can be changed, but remember that there is a 16 character limit per line (including color codes that use 2 characters each).")
-@Examples({"on rightclick on sign:",
-		"	line 2 of the clicked block is \"[Heal]\":",
-		"		heal the player",
-		"	set line 3 to \"%player%\""})
-@Since("1.3")
+@Examples({
+	"on rightclick on sign:",
+		"\tline 2 of the clicked block is \"[Heal]\":",
+			"\t\theal the player",
+		"\tset line 3 to \"%player%\""
+})
+@Since("1.3, INSERT VERSION (all lines, back side, multiple blocks, and Skript's ChatFormat (hex, font, etc))")
+@RequiredPlugins({
+	"Paper 1.16+ or Adventure API installed to use Skript ChatFormat",
+	"Spigot 1.20+ required to use the sign side"
+})
 public class ExprSignText extends SimpleExpression<String> {
+
+	private static final boolean RUNNING_1_20 = Skript.isRunningMinecraft(1, 20);
+
 	static {
+		String addition = RUNNING_1_20 ? "[[on [the] (front|:back) side] of [sign[s]] %blocks%]" : "[of [sign[s]] %blocks%]";
 		Skript.registerExpression(ExprSignText.class, String.class, ExpressionType.PROPERTY,
-				"[the] line %number% [of %block%]", "[the] (1¦1st|1¦first|2¦2nd|2¦second|3¦3rd|3¦third|4¦4th|4¦fourth) line [of %block%]");
+				"[all [[of] the]|the] lines " + addition,
+				RUNNING_1_20 ? "%blocks%'[s] [(front|:back) side] lines" : "%blocks%'[s] lines",
+
+				"[the] line %number% " + addition,
+				"[the] (1¦1st|1¦first|2¦2nd|2¦second|3¦3rd|3¦third|4¦4th|4¦fourth) line[s] " + addition);
```

Comment:
Cannot due to tags being used for `back` and `front` side of signs. Would need a complex compare against matchedPattern.

Reply:
Those shouldn't interfere at all. Your code would need no changes to allow this.

Reply:
Unresolved

Reply:
Parse mark is more efficient and easier to read and implement as said above.

Reply:
using : would still set the parse mark
as I said, you would have to change nothing about your code.

Reply:
Tags are strings, not numbers, you would need to do an un-nessasary cast.
If you were to make the tags reflected based on their name, an enum could be used, but then you would still need to remember the line number as the sign get line argument is a number, another reason why the mark should remain.

Reply:
no lime literally it will set the parse mark
this is how : works! if it's just a number, it will set the parse mark AND add the tag.

Reply:
![image](https://github.com/SkriptLang/Skript/assets/10354869/7a326d45-7fdf-451f-ae67-18c1753f2999)

Reply:
Trust me, I originally started development doing this exact change, made an enum for the pattern similar to inventory enum pattern elements, and went back to this current implementation as it was more efficient and not over complicated.

Reply:
Sorry Lime, the broken pipe symbol is deprecated, we can't be continuing to use it. Please change them to colons.

Reply:
Where is this defined?

Reply:
I don't think it has been written down anywhere concretely, but it is what we've been doing for the past few years. 
As I have said multiple times now, this requires literally no effort on your part except just clicking "commit suggestion". Please do so, or we will do it for you.

Reply:
This is an unprofessional pitty argument in reference to an issue I made https://github.com/SkriptLang/Skript/issues/5489

If you have a genuine reasoning for this change, please state so, otherwise marking as resolved.

Reply:
This is an unprofessional petty argument in reference to an issue I made https://github.com/SkriptLang/Skript/issues/5489

If you have a genuine reasoning for this change, please state so, otherwise marking as resolved.